### PR TITLE
Obsolete serialization of 2 types

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/14068

I made some attempt to fix serialization for each of these types, but they have been sufficiently rewritten that it would need some design work (eg., some of the state was moved to nested or nested-nested types). Given the serializers were marked obsolete in Full framework, I chose to abandon that effort and made them throw rather than be broken.

@stephentoub @davidsh @CIPop

### CODE REVIEW

**Comment by danmosemsft** [comment]
Hmm @weshaggard any idea why I am getting `HttpWebRequestTest.cs(10,36): error CS0234: The type or namespace name 'Formatters' does not exist in the namespace 'System.Runtime.Serialization' (are you missing an assembly reference?) [D:\j\workspace\windows_nt_re---37265eab\src\System.Net.Requests\tests\System.Net.Requests.Tests.csproj]` in CI but not locally? There is even existing use of BinaryFormatter in this same project.

**Comment by danmosemsft** [comment]
@dotnet-bot test this please (why not)

**Comment by jkotas** [comment]
> the type or namespace name 'Formatters'

The test has multiple build flavors in System.Net.Requests.Tests.builds. You can get the error locally by running `msbuild System.Net.Requests.Tests.builds`. The error is coming from the `netstandard1.3` build flavor... .

**Comment by danmosemsft** [comment]
Uh, thank you. I still find this confusing.

File: src/System.Net.Requests/ref/System.Net.Requests.cs [hunk 1]
```diff
@@ -126,6 +126,8 @@ public class FtpWebResponse : WebResponse, IDisposable
     public partial class HttpWebRequest : System.Net.WebRequest, System.Runtime.Serialization.ISerializable
     {
         internal HttpWebRequest() { }
+        [System.ObsoleteAttribute("Serialization is obsoleted for this type.  http://go.microsoft.com/fwlink/?linkid=14202")]
```
Comment (weshaggard) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: Is there a reason you needed to expose it in the ref?

Comment (weshaggard) [inline src/System.Net.Requests/ref/System.Net.Requests.cs]: Nevermind I see it is in netstandard.

File: src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (CIPop) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: @weshaggard is this how we envision testing for all future versions? Each new feature will need to be separated into a new versioned file?

File: src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+using System.IO;
+using System.Net.Http;
+using System.Runtime.Serialization.Formatters.Binary;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Tests
+{
+    public partial class HttpWebRequestTest17
```
Comment (CIPop) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: Class name must be the same with the file name (i.e. HttpWebRequestTest in this case).

File: src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        [Fact]
+        public void Serialize()
```
Comment (CIPop) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: Nit: Test name should be matching our format: <tested_feature>_<shorttestdescription>_<expectedresult>.

File: src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs (hunk 2 above)
Comment (weshaggard) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: Yes for any projects that need to build in multiple configurations then we have been primarily splitting into other files and including the configuration as part of the file name. See https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md#code-file-naming-conventions.

Comment (CIPop) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: The document is giving examples only for non-version pivots. My question is about version pivots such as netstandard1.3, 1.7/2.0 and future v3, v4, v5, etc. 
/cc @karelz @ericstj @stephentoub 

What is the benefit for testing netstandard1.3 and netstandard1.7 at this point? Isn't 1.3 a subset of 1.7? 
Maintaining all backward versions will soon create a very complex set of tests that will be hard to maintain and understand. E.g.:

> Uh, thank you. I still find this confusing.

Comment (weshaggard) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: Those are TargetGroup pivots so they are just another example. 

>What is the benefit for testing netstandard1.3 and netstandard1.7 at this point? Isn't 1.3 a subset of 1.7?

The benefit is so we can still run the majority of tests on platforms that don't yet support netstandard1.7, example UWP. However with that said we are in the process of updating the corefx engineering process (in dev/eng branch) in which case we will essentially only build for the latest version of each target framework (i.e. latest netcoreapp, netstandard, uap, netfx, etc) and once that happens we will no longer have a netstandard1.3 configuration.

Comment (CIPop) [inline src/System.Net.Requests/tests/HttpWebRequestTest.netstandard1.7.cs]: Thanks @weshaggard. That makes sense!

**Comment by danmosemsft** [comment]
@CIPop ok now?

**Comment by danmosemsft** [comment]
Doh - didn't push the right change

**Comment by danmosemsft** [comment]
OK, good now

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop osx debug build and test (https://github.com/dotnet/corefx/issues/14228)

**Pull request merged**